app/models/course/settings/materials_component.rb

```diff
@@ -6,9 +6,10 @@ class Course::Settings::MaterialsComponent
 
   # Initialises the settings adapter
   #
-  # @param [#settings] settings The settings object provided by the settings_on_rails gem.
-  def initialize(settings)
-    @settings = settings
+  # @param [#current_course, #key] context The component that is the context for the settings
+  def initialize(context)
+    @context = context
```

Comment:
Fixed. In the process, I removed `include ActiveModel::Model` because it overrides `SimpleDelegator`'s `initialize` -- this seems safe to do because `ActiveModel::Validations` and `ActiveModel::Conversion` already provide the methods we need.